app/assets/src/components/views/bulk_download/ChooseStep.jsx

```diff
@@ -201,23 +236,54 @@ class ChooseStep extends React.Component {
     let showNoResultsMessage = false;
     let isLoadingSearchOptions = false;
 
-    // Set different props for the dropdown depending on the field type.
-    switch (field.type) {
-      case "file_format":
+    // Handle rendering conditional fields.
+
+    // For the file format field, render a placeholder. This is a special case.
+    const fileFormatConditionalField = CONDITIONAL_FIELDS[0];
+    if (
+      field.type === fileFormatConditionalField.field &&
+      downloadType.type === fileFormatConditionalField.downloadType &&
+      !fileFormatConditionalField.triggerValues.includes(
+        get(
+          [downloadType.type, fileFormatConditionalField.dependentField],
+          selectedFields
+        )
+      )
+    ) {
+      return (
+        <div className={cs.field} key={field.type}>
+          <div className={cs.label}>{field.display_name}:</div>
+          <div className={cs.forcedOption}>.fasta</div>
+          <div className={cs.info}>
+            Note: Only .fasta is available when selecting one taxon.
+          </div>
+        </div>
+      );
+      // For other conditional fields, render nothing.
+    } else {
+      let abortRenderOption = false;
+
+      CONDITIONAL_FIELDS.forEach(conditionalField => {
         if (
-          downloadType.type === READS_NON_HOST_DOWNLOAD_TYPE &&
-          isNumber(get([downloadType.type, "taxa_with_reads"], selectedFields))
+          field.type === conditionalField.field &&
+          downloadType.type === conditionalField.downloadType &&
+          !conditionalField.triggerValues.includes(
+            get(
+              [downloadType.type, conditionalField.dependentField],
+              selectedFields
+            )
+          )
         ) {
-          return (
-            <div className={cs.field} key={field.type}>
-              <div className={cs.label}>{field.display_name}:</div>
-              <div className={cs.forcedOption}>.fasta</div>
-              <div className={cs.info}>
-                Note: Only .fasta is available when selecting one taxon.
-              </div>
-            </div>
-          );
+          abortRenderOption = true;
```

Comment:
since you can `return` on the first `true`, you can actually just use `CONDITIONAL_FIELDS.every()` to break earlier, right?

Reply:
Yes! Thanks for the suggestion.